src/main/java/com/arpnetworking/clusteraggregator/aggregation/AggMessageExtractor.java

```diff
@@ -52,6 +56,20 @@ public String entityId(final Object message) {
                     .append(metricData.getMetric())
                     .append("||")
                     .append(metricData.getPeriod());
+            final ArrayList<Messages.DimensionEntry> sortedDimensions = new ArrayList<>(metricData.getDimensionsList());
+            sortedDimensions.sort(Comparator.comparing(Messages.DimensionEntry::getKey));
```

Comment:
Right! -- sorry, was thinking of the client protocol. What about the intention here? Would it be safe to say consider this a map and validate it as such? That would simplify the ordering and handle the multi-value case nicely.